### ISSUE TITLE
[java] DoNotExtendJavaLangThrowable

### ISSUE THREAD

**Issue by george-007** [pmd/pmd#367]
**Rule Set:**
java - Strict Exception Rules

**Description:**
It would be nice to have a rule ``DoNotExtendJavaLangThrowable`` forbidding direct extension of ``java.lang.Throwable``, ie similar to ``DoNotExtendJavaLangError`` but for ``Throwable``.

**Code Sample demonstrating the issue:**
The following code should trigger a warning:
```java
public NotFoundException extends Throwable {
```

**Comment by jsotuyod** [comment]
@george-007 sounds reasonable. Thanks for the suggestion!
